packages/apps/desktop-viewer-test/src/frontend/extensions/IModelMergeStatusBarItem.tsx

```diff
@@ -0,0 +1,143 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import "./IModelMergeStatusBarItem.scss";
+
+import { InternetConnectivityStatus } from "@bentley/imodeljs-common";
+import {
+  BriefcaseConnection,
+  IModelApp,
+  IModelConnection,
+} from "@bentley/imodeljs-frontend";
+import {
+  CommonStatusBarItem,
+  StageUsage,
+  StatusBarSection,
+  UiItemsProvider,
+} from "@bentley/ui-abstract";
+import { StatusBarItemUtilities } from "@bentley/ui-framework";
+import {
+  getBriefcaseStatus,
+  ModelStatus,
+  useAccessToken,
+  useConnectivity,
+  useIsMounted,
+} from "@itwin/desktop-viewer-react";
+import { SvgUser } from "@itwin/itwinui-icons-react";
+import React, { useCallback, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+
+import { BriefcaseStatus } from "../components/modelSelector";
+
+const MergeStatusBarItem = () => {
+  const [mergeStatus, setMergeStatus] = useState<ModelStatus>();
+  const [connection, setConnection] = useState<BriefcaseConnection>();
+  const accessToken = useAccessToken();
+  const connectivityStatus = useConnectivity();
+  const iModelConnection = useSelector((state: any) => {
```

Comment:
guess you couldnt use the useActiveIModelConnection hook because thats dependent on the view maanger which will be stale/out of synch, so you needed to access it like so

Reply:
I would prefer to use that hook but didn't get the results that I expected. However, that was pre-refactor. I'll try again before merging

Reply:
Works as expected now. I updated. Great catch.